packages/webview_flutter/example/test_driver/webview_flutter_e2e.dart

```diff
@@ -828,6 +828,90 @@ void main() {
+      final Completer<WebViewController> controllerCompleter =
+          Completer<WebViewController>();
+      final Completer<void> pageLoaded = Completer<void>();
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: WebView(
+            key: GlobalKey(),
+            onWebViewCreated: (WebViewController controller) {
+              controllerCompleter.complete(controller);
+            },
+            javascriptMode: JavascriptMode.unrestricted,
+            onPageFinished: (String url) {
+              pageLoaded.complete();
+            },
+            initialUrl: 'https://flutter.dev',
+          ),
+        ),
+      );
+      final WebViewController controller = await controllerCompleter.future;
+      await controller
+          .evaluateJavascript('window.open("https://www.google.com")');
+      await pageLoaded.future;
+      expect(controller.currentUrl(), completion('https://www.google.com/'));
+
+      await controller.goBack();
+      expect(controller.currentUrl(), completion('https://www.flutter.dev'));
+    },
+    skip: !Platform.isAndroid,
+  );
+
+  testWidgets(
+    'javascript does not run in parent window',
+    (WidgetTester tester) async {
+      final String openWindowTest = '''
+        <!DOCTYPE html><html>
+        <head><title>Resize test</title>
+          <script>
+            var iframeLoaded = false;
+            function onLoad() {
+              window.open('javascript:
+                var elem = document.createElement("p");
+                elem.innerHTML = "<b>Executed JS in parent origin: " + window.location.origin + "</b>";
+                document.body.append(elem);
+                alert("XSS in doc.domain: " + document.domain + ", win.origin: " + window.location.origin)');
+            }
+          </script>
+        </head>
+        <body onload="onLoad();" bgColor="blue">
+        </body>
+        </html>
+      ''';
+      final String openWindowTestBase64 =
+          base64Encode(const Utf8Encoder().convert(openWindowTest));
+      final Completer<WebViewController> controllerCompleter =
+          Completer<WebViewController>();
+
+      await tester.pumpWidget(
+        Directionality(
+          textDirection: TextDirection.ltr,
+          child: WebView(
+            key: GlobalKey(),
+            onWebViewCreated: (WebViewController controller) {
+              controllerCompleter.complete(controller);
+            },
+            javascriptMode: JavascriptMode.unrestricted,
+            initialUrl:
+                '<iframe onload="iframeLoaded = true;" src="data:text/html;charset=utf-8;base64,$openWindowTestBase64"</iframe>',
+          ),
+        ),
+      );
+
+      final WebViewController controller = await controllerCompleter.future;
+      final String result = await controller.evaluateJavascript(
+        'iframeLoaded && document.querySelector("p") && document.querySelector("p").textContent',
```

Comment:
this might be empty if `iframeLoaded` is `false`. I was thinking something like this:
```js
expect(controller.evaluateJavascript('iframeLoaded'), completion('true'));

expect(controller.evaluateJavascript('document.querySelector("p") && document.querySelector("p").textContent'), isEmpty);
```

Reply:
Actually, when `iframeLoaded` was false, it returned `'false'`, but I understand what you mean. done